foundations/foundation-vertx/src/main/java/org/apache/servicecomb/foundation/vertx/metrics/DefaultHttpClientMetrics.java

```diff
@@ -133,7 +144,18 @@ public void disconnected(Object webSocketMetric) {
 
   @Override
   public DefaultHttpSocketMetric connected(SocketAddress remoteAddress, String remoteName) {
-    return new DefaultHttpSocketMetric(null);
+
+    DefaultHttpSocketMetric socketMetric = new DefaultHttpSocketMetric(null);
+    try {
+      DefaultHttpClientMetrics clientMetrics = (DefaultHttpClientMetrics) ((HttpClientImpl) client).getMetrics();
+      DefaultClientEndpointMetric clientEndpointMetric = clientMetrics.clientEndpointMetricManager
+          .getClientEndpointMetricMap().get(remoteAddress);
+      // set endPointMetric when use http2
+      socketMetric.setEndpointMetric(clientEndpointMetric);
+    } catch (Exception e) {
+      LOGGER.warn("if you use http2, there may cause a null pointer exception. {}/{}", remoteAddress, remoteName);
```

Comment:
when will this exception happened?

Reply:
because I have cast client to DefaultHttpClientMetrics  and invoke the  method getMetrics().  I'm afraid that the method getMetrics() may return null, so I just catch it for safety.   But now, it seems impossible, I will change it

Reply:
because I have cast client to DefaultHttpClientMetrics and invoke the method getMetrics(). I'm afraid that the method getMetrics() may return null, so I just catch it for safety. But now, it seems impossible, I will modify it